src/build.d

```diff
@@ -418,6 +420,91 @@ alias runDmdUnittest = memoize!(function()
     return new DependencyRef(dep);
 });
 
+/// Compiles the C++ frontend test files
+alias cxxFrontend = memoize!(function()
+{
+    Dependency dep;
+    with (dep)
+    {
+        name = "cxx-frontend";
+        description = "Build the C++ frontend";
+        msg = "(CXX) CXX-FRONTEND";
+
+        sources = srcDir.buildPath("tests", "cxxfrontend.c") ~ .sources.sources ~ .sources.root;
```

Comment:
for consistency with everything else, use `.chain(<args>).array`

Reply:
Can do, but that makes it much more verbose:
```suggestion
- sources = srcDir.buildPath("tests", "cxxfrontend.c") ~ .sources.sources ~ .sources.root;

+ sources = [srcDir.buildPath("tests", "cxxfrontend.c")].chain(.sources.sources, .sources.root).array;
or
+ sources = chain([srcDir.buildPath("tests", "cxxfrontend.c")], .sources.sources, .sources.root).array;
```

Reply:
Can do, but that makes it much more verbose:
```suggestion
- sources = srcDir.buildPath("tests", "cxxfrontend.c") ~ .sources.sources ~ .sources.root;

+ sources = [srcDir.buildPath("tests", "cxxfrontend.c")].chain(.sources.sources, .sources.root).array;
or
+ sources = chain([srcDir.buildPath("tests", "cxxfrontend.c")], .sources.sources, .sources.root).array;
```

Reply:
Slightly off topic, but it would be nice to use something like c++'s std::path instead of strings (which overloads / ) to improve the readability of `sources`, `target`, ...

Reply:
Like Vibe.d's `NativePath`?

https://github.com/dlang/dub/blob/master/source/dub/internal/vibecompat/inet/path.d#L31

Phobos is in feature-freeze and we can't rely on dub packages here. So the only option is make it part of `build.d` which is already quite big.

Reply:
> for consistency with everything else, use .chain(<args>).array

IIRC `chain` was initially only used where there were ranges. Even if it is now used more prominently, I don't think that we should be pedantic about it. It's really just useful if you work with non-arrays.